### ISSUE TITLE
Unable to make remark plugins work

### ISSUE THREAD

**Issue by pbteja1998** [hashicorp/next-mdx-remote#74]
Hi
I have the following remark plugins in my config. But I am unable to make them work. The only plugin that works is `remark-prism` plugin. Can you please tell me if I am doing something wrong here?

```
remarkPlugins: [
    require('remark-prism'),
    require('remark-autolink-headings'),
    require('remark-slug'),
    require('remark-code-titles'),
],
```

Among these, only `remark-prism` plugin works. I am unable to make the rest of these work.

Thanks in advance.

**Comment by jescalan** [comment]
Hey @pbteja1998 - can you keep working on this a little more? What I'd do next:

- Move another plugin to first in the stack. Does it work now?
- Remove everything else except one plugin. Does the one work now when it didn't before?

See if you can narrow down the diagnosis of what's going wrong 😀

**Comment by stevenschmatz** [comment]
Hey @pbteja1998, I recently added `remark-code-titles` to my project with `next-mdx-remote` and happy to field any questions.

**Comment by pbteja1998** [comment]
I was able to make this work. I had to put (`remark-code-titles` above `remark-prism`) and (`remark-slug` above `remark-autolink-headings`). 

`remark-autolink-headings` did not work for dynamic routes because the URL is getting converted to something like `http://localhost:300/[slug]#section-slug` instead of `http://localhost:3000/getting-started#section-slug`. It is not taking the actual URL. 

I had to tweak the `remark-autolink-headings` package a bit and made it accept baseUrl as one of the options. I then generated the URL in the following way inside the package.
`${baseUrl}#${sectionSlug}`
